supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/Preprocessor.ts

```diff
@@ -0,0 +1,320 @@
+                node.value.body.body.forEach((statement) => {
+                    if (statement.type === 'ExpressionStatement' && statement.expression.type === 'CallExpression' && statement.expression.callee.type === 'Super') {
+                        const superConstructorCall = statement.expression;
+                        const superConstructorCallParameters = superConstructorCall.arguments;
+                        const superConstructorCallParameterValues: string[] = [];
+                        superConstructorCallParameters.forEach((parameter) => {
+                            superConstructorCallParameterValues.push(this.evaluateCallExpressionArgument(parameter));
+                        });
+                        result.push({arguments: superConstructorCallParameterValues});
+                    }
+                })
+            }
+        });
+
+        return result;
+    }
+
+    /**
+     * Evaluates a call expression argument to find its corresponding value.
+     * This method checks the type of the call expression argument and processes it accordingly.
+     * If the argument is an identifier, it evaluates the identifier to find its value.
+     * If the argument is a binary expression, it evaluates the binary expression to find its value.
+     * If the argument is a literal, it converts the literal value to a string.
+     * The method returns the evaluated value as a string, or an empty string if no value is found.
+     *
+     * @param callExpressionArgument - The call expression argument to be evaluated.
+     * @returns The evaluated value of the call expression argument as a string, or an empty string if no value is found.
+     */
+    evaluateCallExpressionArgument(callExpressionArgument: TSESTree.CallExpressionArgument) {
+        let result = '';
+        if (callExpressionArgument.type === 'Identifier') {
+            result = this.evaluateIdentifierArgument(callExpressionArgument) ?? '';
+        } else if (callExpressionArgument.type === 'BinaryExpression') {
+            result = this.evaluateBinaryExpression(callExpressionArgument);
+        } else if (callExpressionArgument.type === 'Literal' && callExpressionArgument.value) {
+            result = callExpressionArgument.value.toString();
+        }
+        return result? result : '';
+    }
+
+    /**
+     * Evaluates an identifier argument to find its corresponding value.
+     * This method checks if the identifier name exists in the `memberVariables` map.
+     * If it exists, it returns the value associated with the identifier name.
+     * If it does not exist, it returns an empty string.
+     *
+     * @param identifier - The identifier whose value is to be evaluated.
+     * @returns The value of the identifier if found; otherwise, an empty string.
+     */
+    evaluateIdentifierArgument(identifier: TSESTree.Identifier) {
```

Comment:
```suggestion
    evaluateIdentifierArgument(identifier: TSESTree.Identifier): string {
```

or `string | undefined` for the current implementation